test/autorest/generated/httpinfrastructuregroup/httpredirects.go

```diff
@@ -0,0 +1,693 @@
+	// Post307 - Post redirected with 307, resulting in a 200 after redirect
+	Post307(ctx context.Context) (*HTTPRedirectsPost307Response, error)
+	// Put301 - Put true Boolean value in request returns 301.  This request should not be automatically redirected, but should return the received 301 to the caller for evaluation
+	Put301(ctx context.Context) (*HTTPRedirectsPut301Response, error)
+	// Put307 - Put redirected with 307, resulting in a 200 after redirect
+	Put307(ctx context.Context) (*HTTPRedirectsPut307Response, error)
+}
+
+// httpRedirectsOperations implements the HTTPRedirectsOperations interface.
+type httpRedirectsOperations struct {
+	*Client
+}
+
+// Delete307 - Delete redirected with 307, resulting in a 200 after redirect
+func (client *httpRedirectsOperations) Delete307(ctx context.Context) (*HTTPRedirectsDelete307Response, error) {
+	req, err := client.delete307CreateRequest()
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.p.Do(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	result, err := client.delete307HandleResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// delete307CreateRequest creates the Delete307 request.
+func (client *httpRedirectsOperations) delete307CreateRequest() (*azcore.Request, error) {
+	urlPath := "/http/redirect/307"
+	u, err := client.u.Parse(urlPath)
+	if err != nil {
+		return nil, err
+	}
+	req := azcore.NewRequest(http.MethodDelete, *u)
+	return req, req.MarshalAsJSON(true)
+}
+
+// delete307HandleResponse handles the Delete307 response.
+func (client *httpRedirectsOperations) delete307HandleResponse(resp *azcore.Response) (*HTTPRedirectsDelete307Response, error) {
+	if !resp.HasStatusCode(http.StatusOK) {
+		return nil, newError(resp)
+	}
+	result := HTTPRedirectsDelete307Response{RawResponse: resp.Response}
+	if val := resp.Header.Get("Location"); val != "" {
+		result.Location = &val
+	}
+	return &result, nil
+}
+
+// Get300 - Return 300 status code and redirect to /http/success/200
+func (client *httpRedirectsOperations) Get300(ctx context.Context) (*HTTPRedirectsGet300Response, error) {
+	req, err := client.get300CreateRequest()
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.p.Do(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	result, err := client.get300HandleResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// get300CreateRequest creates the Get300 request.
+func (client *httpRedirectsOperations) get300CreateRequest() (*azcore.Request, error) {
+	urlPath := "/http/redirect/300"
+	u, err := client.u.Parse(urlPath)
+	if err != nil {
+		return nil, err
+	}
+	req := azcore.NewRequest(http.MethodGet, *u)
+	return req, nil
+}
+
+// get300HandleResponse handles the Get300 response.
+func (client *httpRedirectsOperations) get300HandleResponse(resp *azcore.Response) (*HTTPRedirectsGet300Response, error) {
+	if !resp.HasStatusCode(http.StatusOK) {
```

Comment:
This should be `if !resp.HasStatusCode(http.StatusOK, http.StatusMultipleChoices)`

Reply:
I added the status code to the list of status codes we can add in this function call, but the modeler is only returning this as a single status code 200 OK for this response. Will confirm if this is expected behavior with the autorest crew.

Reply:
My earlier comment was from the track 1 codegen.  It could be the case that this has changed for track 2.